deps/src/main/java/com/microsoft/azure/sdk/iot/deps/serializer/RegisterManagerParser.java

```diff
@@ -111,7 +111,7 @@ protected boolean setDeviceId(String deviceId) throws IllegalArgumentException
         validateDeviceManager(deviceId, null, null);
 
         /* Codes_SRS_TWIN_21_159: [The updateDeviceManager shall replace the `deviceId` by the provided one.] */
-        if((this.deviceId == null) || (deviceId == null) || (!this.deviceId.equals(deviceId)))
+        if((this.deviceId == null) || (!this.deviceId.equals(deviceId)))
```

Comment:
is this (deviceId == null) validation done in validateDeviceManager()?

Reply:
It is checked for null earlier than this line, yeah